localstack-core/localstack/services/lambda_/event_source_listeners/kinesis_event_source_listener.py

```diff
@@ -147,6 +147,7 @@ def _transform_records(self, raw_records: list[dict]) -> list[dict]:
             record_payload = {}
             for key, val in record.items():
                 record_payload[first_char_to_lower(key)] = val
+            # TODO: look into CBOR parsing issues (we had demos by Alex and Simon)
```

Comment:
I don't think there's anything CBOR related to consider here. I would assume the ESM listener only uses `botocore` to communicate with Kinesis? If so, there's not CBOR involved (`botocore` doesn't support CBOR at all).

> AWS requires millisecond precision, but the timestamp has to be in seconds with the milliseconds represented by the fraction part of the float

Since you are mentioning CBOR above, it might make sense to clarify that here:
- This comment is true for the JSON format of Kinesis.
- This comment is correct for the specification of CBOR
- But this is not how Kinesis CBOR is implemented (they use integer epoch milliseconds instead of the floating point epoch seconds defined in the spec).

Is it really necessary to transform this here manually? The parser / serializer should take care of transforming datetime records anyway?

Reply:
>  I would assume the ESM listener only uses botocore to communicate with Kinesis?

Yes indeed. (We use the Amazon KCL library for Python in other places such as [here](localstack.utils.kinesis.kinesis_connector.KinesisProcessor), which uses a Java-based consumer library under the hood, but not for ESM.)

> If so, there's not CBOR involved (botocore doesn't support CBOR at all).

Thanks for the clarification. Glad to hear it's a non-issue here 😎 

> Is it really necessary to transform this here manually? The parser / serializer should take care of transforming datetime records anyway?

That's the old implementation, but it seems necessary also in the the new `kinesis_poller` implementation. For DynamoDB, the test `aws.services.lambda_.event_source_mapping.test_lambda_integration_dynamodbstreams.TestDynamoDBEventSourceMapping.test_dynamodb_event_filter` demands an additional conversation from the integer timestamp into float, implemented in the `dynamodb_poller`:
```
if creation_time := dynamodb.get("ApproximateCreationDateTime"):
    # validated by TestDynamoDBEventSourceMapping.test_dynamodb_event_filter
    dynamodb["ApproximateCreationDateTime"] = float(creation_time.timestamp())
```